### PR TITLE
Fix: ssm prefix custom cdk policy

### PR DESCRIPTION
**Pull request by dlpzx**
### Feature or Bugfix
- Bugfix

### Detail
As part of the environment stack we deploy some SSM parameters prefixed with `dataall` prefix in all cases. The custom DataallCustomCDKPolicy provided to use in the `cdk bootstrap` command restricts SSM permissions to resource-prefixed parameters. As a result, when using an environment with a resource prefix different from the default `dataall` the stack fails to create and to delete because it cannot create or delete those SSM parameters.

- The first commit adds the generic `dataall` SSM permisssions to the custom policy. It open the permissions slightly
- The second and third commits rename the SSM parameters to use the resource prefix

Each commit would solve the issue on its own, so we don't really need both. However, there are arguments to keep both. Having generic permissions to dataall-SSM parameters is restrictive and might be useful for other cases when the toolkit is used. It is good to keep all created resources prefixed with the same prefix, so that users can easily track which resources belong to a data.all environment.

The only issue is that for number 2, we need users to update the environment stacks before creating more datasets (add in release notes)

### Relates
- v2.1 release

### Security
Please answer the questions below briefly where applicable, or write `N/A`. Based on
[OWASP 10](https://owasp.org/Top10/en/).

- Does this PR introduce or modify any input fields or queries - this includes
fetching data from storage outside the application (e.g. a database, an S3 bucket)?
  - Is the input sanitized?
  - What precautions are you taking before deserializing the data you consume?
  - Is injection prevented by parametrizing queries?
  - Have you ensured no `eval` or similar functions are used?
- Does this PR introduce any functionality or component that requires authorization?
  - How have you ensured it respects the existing AuthN/AuthZ mechanisms?
  - Are you logging failed auth attempts?
- Are you using or adding any cryptographic features?
  - Do you use a standard proven implementations?
  - Are the used keys controlled by the customer? Where are they stored?
- Are you introducing any new policies/roles/users?
  - Have you used the least-privilege principle? How?

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Comment by dlpzx** [comment]
Tested on AWS deployment. I wanted to know your opinion @noah-paige:
- Should we remove the permission to `dataall-` SSM params?
- Should we go ahead with the renaming? or is it an unnecessary change?
I do not have an strong opinion (only slight opinions) on it :)

File: backend/dataall/modules/datasets/cdk/dataset_custom_resources_extension.py [hunk 1]
```diff
@@ -88,14 +88,14 @@ def extent(setup: EnvironmentSetup):
             setup,
             'LakeformationDefaultSettingsCustomeResourceFunctionArn',
             string_value=lf_default_settings_custom_resource.function_arn,
-            parameter_name=f'/dataall/{setup.environment().environmentUri}/cfn/lf/defaultsettings/lambda/arn',
```
Comment (noah-paige) [inline backend/dataall/modules/datasets/cdk/dataset_custom_resources_extension.py:91]: Minor formatting - should we jsut define `_env = setup.environment()` and then call the `_env.resourcePrefix` and `_env.environmentUri` all the subsequent times?

Comment (dlpzx) [reply backend/dataall/modules/datasets/cdk/dataset_custom_resources_extension.py:91]: Done :)

**Comment by dlpzx** [comment]
Re-tested with the latest changes

**Pull request merged**